### PR TITLE
[AIRFLOW-4132] Set default connection ID for all operators

### PR DESCRIPTION
**Pull request by mik-laj**
Make sure you have checked _all_ steps below.

### Jira

  - https://issues.apache.org/jira/browse/AIRFLOW-4132

### Description

Some operators do not have a default connection ID set, which makes their use more difficult. This PR introduces the default ID for all operators.

### Tests

- [ ] My PR adds the following unit tests __OR__ does not need testing for this extremely good reason:

### Commits

- [ ] My commits all reference Jira issues in their subject lines, and I have squashed multiple commits if they address the same issue. In addition, my commits follow the guidelines from "[How to write a good git commit message](http://chris.beams.io/posts/git-commit/)":
  1. Subject is separated from body by a blank line
  1. Subject is limited to 50 characters (not including Jira issue reference)
  1. Subject does not end with a period
  1. Subject uses the imperative mood ("add", not "adding")
  1. Body wraps at 72 characters
  1. Body explains "what" and "why", not "how"

### Documentation

- [ ] In case of new functionality, my PR adds documentation that describes how to use it.
  - When adding new operators/hooks/sensors, the autoclass documentation generation needs to be added.
  - All the public functions and the classes in the PR contain docstrings that explain what it does
  - If you implement backwards incompatible changes, please leave a note in the [Updating.md](https://github.com/apache/airflow/blob/master/UPDATING.md) so we can assign it to a appropriate release

### Code Quality

- [ ] Passes `flake8`

### CODE REVIEW

File: airflow/contrib/operators/azure_container_instances_operator.py [hunk 1]
```diff
@@ -107,9 +107,24 @@ class AzureContainerInstancesOperator(BaseOperator):
     template_fields = ('name', 'environment_variables')
 
     @apply_defaults
-    def __init__(self, ci_conn_id, registry_conn_id, resource_group, name, image, region,
-                 environment_variables=None, volumes=None, memory_in_gb=None, cpu=None,
-                 command=None, remove_on_error=True, fail_if_exists=True, *args, **kwargs):
+    def __init__(
```
Comment (potiuk) [inline airflow/contrib/operators/azure_container_instances_operator.py]: There is a change in sequence of positional parameters and it's likely someone is using it already. I am not sure this is a good idea to change it in (I assume) already released operator?

File: airflow/contrib/operators/jenkins_job_trigger_operator.py [hunk 2]
```diff
@@ -105,8 +105,8 @@ class JenkinsJobTriggerOperator(BaseOperator):
 
     @apply_defaults
     def __init__(self,
-                 jenkins_connection_id,
                  job_name,
+                 jenkins_connection_id='jenking_default',
```
Comment (potiuk) [inline airflow/contrib/operators/jenkins_job_trigger_operator.py]: Same here. Sequence of parameters is changed.

File: airflow/contrib/operators/oracle_to_azure_data_lake_transfer.py [hunk 3]
```diff
@@ -62,10 +62,10 @@ class OracleToAzureDataLakeTransfer(BaseOperator):
     def __init__(
             self,
             filename,
-            azure_data_lake_conn_id,
             azure_data_lake_path,
-            oracle_conn_id,
             sql,
+            oracle_conn_id='oracle_default',
```
Comment (potiuk) [inline airflow/contrib/operators/oracle_to_azure_data_lake_transfer.py]: Same here.

**Review by potiuk — COMMENTED** [review]
Looks good. But I think we should be more backwards compatible (or at least mention in the doc that we aren't in a few places.

**Comment by mik-laj** [comment]
Done by community

**Pull request closed without merging**